src/themes/components/modal.ts

```diff
@@ -0,0 +1,116 @@
+import { mode } from '@chakra-ui/theme-tools';
+import {
+  createMultiStyleConfigHelpers,
+  defineStyle,
+} from '@chakra-ui/styled-system';
+import { modalAnatomy as parts } from '@chakra-ui/anatomy';
+
+import { opacity } from '../shared/opacity';
+
+const { defineMultiStyleConfig, definePartsStyle } =
+  createMultiStyleConfigHelpers(parts.keys);
+
+const baseStyleOverlay = defineStyle({
```

Comment:
I believe most of those styles are not needed and usually we try to avoid creating the "CSS" files and use the style props directly on the component. Could you try to do that? Thanks!

Reply:
The modal is not working correctly without those styles in those file format. That was the issue that struggle with. I was trying to avoid doing this but the modal would just not work properly. I think it's something related to our setup of theming in chakra ui. Without exporting it like this modal will not work, it's default behaviour is overwritten by our theme.